### PR TITLE
merge code from devel android to devel

### PR DESCRIPTION
**Pull request by bachlx-bitmark**
update code relate to android, 
upgrade react native version, 
update using jwt for profile server, ifttt server and file courier server.

### CODE REVIEW

File: android/app/src/main/java/com/bitmark/registry/modules/navigation/NavigationModule.java [hunk 1]
```diff
@@ -53,12 +60,92 @@ public void onActivityResult(Activity activity, int requestCode, int resultCode,
...
+            @Override
+            public void onNewIntent(Intent intent) {
+
+            }
+        });
+    }
+
+    @ReactMethod
+    public void browseMedia(String type, Promise promise) throws NativeModuleException {
+        if (!type.equals("photo") && !type.equals("video"))
```
Comment (phieudu241) [inline android/app/src/main/java/com/bitmark/registry/modules/navigation/NavigationModule.java]: Should be:
`
!"photo".equals(type) && !"video".equals(type)
`

Check the rest code for: `"XXX".equals(type)`

File: android/app/src/main/java/com/bitmark/registry/utils/MediaUtils.java [hunk 2]
```diff
@@ -33,166 +34,192 @@ public static boolean isExternalStorageDocument(Uri uri) {
...
 
-                final String docId = DocumentsContract.getDocumentId(uri);
-                final String[] split = docId.split(":");
-                final String type = split[0];
+                        callback.onSuccess(
+                                Environment.getExternalStorageDirectory() + "/" + split[1]);
+                    } else {
 
-                if ("primary".equalsIgnoreCase(type)) {
+                        String[] proj = {MediaStore.Images.Media.DATA};
```
Comment (phieudu241) [inline android/app/src/main/java/com/bitmark/registry/utils/MediaUtils.java]: The variable name `proj` stands for what?

File: android/app/src/main/java/com/bitmark/registry/utils/MediaUtils.java [hunk 3]
```diff
@@ -33,166 +34,192 @@ public static boolean isExternalStorageDocument(Uri uri) {
...
+    private static String getFileNameFromUri(Context context, Uri uri) {
+        DocumentFile documentFile = DocumentFile.fromSingleUri(context, uri);
+        if (documentFile == null) throw new IllegalArgumentException("Invalid uri");
+        String name = documentFile.getName();
+        String ext = getFileExt(documentFile.getType());
+        if (name != null) {
+            if (!name.contains(".") | (ext != null && name.contains(".") && !name
+                    .substring(name.lastIndexOf("."))
+                    .contains(ext)))
+                // The file extension is incorrect, change to correct extension from mimetype
```
Comment (phieudu241) [inline android/app/src/main/java/com/bitmark/registry/utils/MediaUtils.java]: The condition can be re-written by:
`
if (!name.contains(".") || (ext != null && !name
                    .substring(name.lastIndexOf("."))
                    .contains(ext)))
`

File: android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            final int badge = data.optInt("badge", -1);
+            if (badge >= 0) {
+                ApplicationBadgeHelper.INSTANCE.setApplicationIconBadgeNumber(this, badge);
+            }
+        }
+
+        Handler handler = new Handler(Looper.getMainLooper());
+        handler.post(() -> {
+            // Construct and load our normal React JS code bundle
+            ReactInstanceManager mReactInstanceManager = ((ReactApplication) getApplication())
```
Comment (phieudu241) [inline android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java]: What does `m` character mean in the variable name `mReactInstanceManager`?

File: android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+            case "notification_intercom_new_messages":
+                return context.getString(R.string.notification_intercom_new_messages);
+            case "notification_tracking_transfer_confirmed":
+                return context.getString(R.string.notification_tracking_transfer_confirmed);
+            case "notification_transfer_confirmed_sender":
+                return context.getString(R.string.notification_transfer_confirmed_sender);
+            case "notification_transfer_confirmed_receiver":
+                return context.getString(R.string.notification_transfer_confirmed_receiver);
+            case "notification_transfer_request":
+                return context.getString(R.string.notification_transfer_failed);
```
Comment (phieudu241) [inline android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java]: Should change `notification_transfer_failed ` to `notification_transfer_request`

File: android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+            case "notification_transfer_failed":
+                return context.getString(R.string.notification_transfer_request);
```
Comment (phieudu241) [inline android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java]: Should change `notification_transfer_request` to `notification_transfer_failed`

File: android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    private boolean isApplicationInForeground() {
+        ActivityManager activityManager = (ActivityManager) this.getSystemService(ACTIVITY_SERVICE);
+        List<ActivityManager.RunningAppProcessInfo> processInfos = activityManager
+                .getRunningAppProcesses();
+        if (processInfos != null) {
+            for (ActivityManager.RunningAppProcessInfo processInfo : processInfos) {
+                if (processInfo.processName.equals(getApplication().getPackageName())) {
+                    if (processInfo.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND) {
+                        for (String d : processInfo.pkgList) {
+                            return true;
```
Comment (phieudu241) [inline android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java]: Should check the size of `processInfo.pkgList` then return

File: src/processors/data-processor.js [hunk 8]
```diff
@@ -676,16 +696,19 @@ const doProcessIncomingClaimRequest = async (incomingClaimRequest, isAccept) =>
...
+  console.log('doIssueIftttData run 1', iftttBitmarkFile);
   let folderPath = FileUtil.CacheDirectory + '/Bitmark-IFTTT';
   await FileUtil.mkdir(folderPath);
   let filename = iftttBitmarkFile.assetInfo.filePath.substring(iftttBitmarkFile.assetInfo.filePath.lastIndexOf("/") + 1, iftttBitmarkFile.assetInfo.filePath.length);
   let filePath = folderPath + '/' + filename;
-  let signatureData = await CommonModel.doCreateSignatureData();
-  let downloadResult = await IftttModel.downloadBitmarkFile(CacheData.userInformation.bitmarkAccountNumber, signatureData.timestamp, signatureData.signature, iftttBitmarkFile.id, filePath);
+  console.log('doIssueIftttData run 2', filePath);
+  let downloadResult = await IftttModel.downloadBitmarkFile(CacheData.jwt, iftttBitmarkFile.id, filePath);
+  console.log('doIssueIftttData run 3', downloadResult);
```
Comment (phieudu241) [inline src/processors/data-processor.js]: Should remove some unnecessary logs

File: src/views/components/home/index.js [hunk 9]
```diff
@@ -70,22 +71,44 @@ export class UserRouterComponent extends Component {
     EventEmitterService.on(EventEmitterService.events.APP_RECEIVED_NOTIFICATION, this.handerReceivedNotification, ComponentName);
   }
 
+  checkNotificationProcessing(eventName, id) {
```
Comment (phieudu241) [inline src/views/components/home/index.js]: Should rename `checkNotificationProcessing` to `checkIfNotificationProcessing`

File: src/views/components/home/properties/local-issuance/issuance-options/issuance-options.component.js [hunk 10]
```diff
@@ -38,6 +51,21 @@ export class PrivateIssuanceOptionsComponent extends React.Component {
     });
   }
 
+  async onChoosePhotoVide() {
```
Comment (phieudu241) [inline src/views/components/home/properties/local-issuance/issuance-options/issuance-options.component.js]: Should rename `onChoosePhotoVide` to `onChooseVideo`

File: android/app/src/main/java/com/bitmark/registry/utils/MediaUtils.java (hunk 2 above)
Comment (jollyjoker992) [reply android/app/src/main/java/com/bitmark/registry/utils/MediaUtils.java]: It's how I named it for `projection`.

File: android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java (hunk 4 above)
Comment (jollyjoker992) [reply android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java]: `NativeFirebaseMessagingService.java` is just a wrapper class of `RNPushNotificationListenerService` comes from the `react-native-push-notification` library, so I keep all of thing that is defined by that library and just modify what we need to do in expectation.

File: android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java (hunk 7 above)
Comment (jollyjoker992) [reply android/app/src/main/java/com/bitmark/registry/utils/notification/NativeFirebaseMessagingService.java]: Ref: https://github.com/bitmark-inc/bitmark-mobile-registry/pull/7#discussion_r266745410

File: src/views/components/home/properties/local-issuance/issuance-options/issuance-options.component.js (hunk 10 above)
Comment (bachlx-bitmark) [reply src/views/components/home/properties/local-issuance/issuance-options/issuance-options.component.js]: done

File: src/views/components/home/index.js (hunk 9 above)
Comment (bachlx-bitmark) [reply src/views/components/home/index.js]: done

File: src/processors/data-processor.js (hunk 8 above)
Comment (bachlx-bitmark) [reply src/processors/data-processor.js]: done

**Pull request merged**